service/src/main/kotlin/fi/espoo/evaka/reservations/AttendanceReservationController.kt

```diff
@@ -367,17 +367,23 @@ class AttendanceReservationController(
                                     FiniteDateRange(it.startDate, it.endDate).includes(date)
                                 } ?: return@mapNotNull null
                             val reservationTimes = reservations[date] ?: emptyList()
-                            val absence = absences.firstOrNull { it.date == date }
+                            val daysAbsences = absences.filter { it.date == date }
                             val dailyServiceTime =
                                 dailyServiceTimes.firstOrNull {
                                     it.times.validityPeriod.includes(date)
                                 }
+                            val absenceCategories = placement.type.absenceCategories()
+                            val isFullDayAbsent =
+                                daysAbsences.map { it.category }.toSet() == absenceCategories
+
                             ConfirmedRangeDate(
                                 date = date,
                                 scheduleType =
                                     placement.type.scheduleType(date, clubTerms, preschoolTerms),
                                 reservations = reservationTimes,
-                                absenceType = absence?.absenceType,
+                                absenceType =
+                                    if (isFullDayAbsent) daysAbsences.firstOrNull()?.absenceType
```

Comment:
Would it make sense to add some kind of sorting before `firstOrNull` to make this deterministic? Usually if there's two absences for a day then both have the same type, but not necessarily. I think there was some other similar case where the absence with billable category was chosen to have the precedence.

Reply:
I tried to change `absenceType` to a boolean field `isAbsent`, but the same data is used for form state initialization in `MarkReservations.tsx`, and things got complicated. So I changed it now to prefer the billable category.